pinot-common/src/main/java/com/linkedin/pinot/pql/parsers/pql2/ast/GroupByAstNode.java

```diff
@@ -36,10 +37,9 @@ public void updateBrokerRequest(BrokerRequest brokerRequest) {
         // List of expression contains columns as well as expressions to maintain ordering of group by columns.
         groupBy.addToExpressions(groupByColumnName);
       } else {
-        FunctionCallAstNode functionCallAstNode = (FunctionCallAstNode) astNode;
-
-        // Remove all white-space until we start compiling expressions on broker side.
-        groupBy.addToExpressions(functionCallAstNode.getExpression());
+        // Compile to standard expression string
+        String expression = ((FunctionCallAstNode) astNode).getExpression();
+        groupBy.addToExpressions(TransformExpressionTree.compileToExpressionTree(expression).toString());
```

Comment:
why are we parsing it here. Is this data structure serializable? How will this backward compatible?

Reply:
Here we just check whether the expression string can be compiled, and put the standard equivalent expression string into the group-by expressions. This way we can avoid invalid query and ensure the "groupByColumns" in the broker response is standard. Will update the comments